linux_os/guide/system/accounts/accounts-pam/locking_out_password_attempts/accounts_passwords_pam_faillock_unlock_time/ansible/shared.yml

```diff
@@ -5,76 +5,96 @@
-    new_control: required
-    new_module_path: pam_faillock.so
-    module_arguments: 'preauth
-        silent
-        unlock_time={{ var_accounts_passwords_pam_faillock_unlock_time }}'
-    state: before
-  loop:
-    - system-auth
-    - password-auth
+- name: Check if system relies on authconfig
+  ansible.builtin.stat:
+    path: /usr/sbin/authconfig
+  register: result_authconfig_check
+
+- name: Check the presence of /etc/security/faillock.conf file
+  ansible.builtin.stat:
+    path: /etc/security/faillock.conf
+  register: result_faillock_conf_check
+
+- name: Ensure the pam_faillock.so unlock_time parameter in /etc/security/faillock.conf
+  ansible.builtin.lineinfile:
+    path: /etc/security/faillock.conf
+    regexp: ^\s*unlock_time\s*=
+    line: unlock_time = {{ var_accounts_passwords_pam_faillock_unlock_time }}
+    state: present
+  when:
+    - result_faillock_conf_check.stat.exists
 
-- name: Add unlock_time argument to pam_faillock preauth
-  pamd:
-    name: "{{ item }}"
-    type: auth
-    control: required
-    module_path: pam_faillock.so
-    module_arguments: 'preauth
-        silent
-        unlock_time={{ var_accounts_passwords_pam_faillock_unlock_time }}'
-    state: args_present
+- name: Check if pam_faillock.so is already enabled
+  ansible.builtin.lineinfile:
+    path: /etc/pam.d/system-auth
+    regexp: .*auth.*pam_faillock.so.*
+    state: absent
+  check_mode: yes
+  changed_when: false
+  register: result_pam_faillock_enabled
+
+- name: Ensure the pam_faillock.so preauth parameters are not present in the pam files
+  ansible.builtin.lineinfile:
+    path: '{{ item }}'
+    backrefs: true
+    regexp: (.*pam_faillock.so preauth).*$
+    line: \1 silent
+    state: present
   loop:
-    - system-auth
-    - password-auth
+    - /etc/pam.d/system-auth
+    - /etc/pam.d/password-auth
+  when:
+    - result_faillock_conf_check.stat.exists
+    - result_pam_faillock_enabled.found
 
-- name: Add auth pam_faillock authfail unlock_interval after pam_unix.so
-  pamd:
-    name: "{{ item }}"
-    type: auth
-    control: sufficient
-    module_path: pam_unix.so
-    new_type: auth
-    new_control: '[default=die]'
-    new_module_path: pam_faillock.so
-    module_arguments: 'authfail
-        unlock_time={{ var_accounts_passwords_pam_faillock_unlock_time }}'
-    state: after
+- name: Ensure the pam_faillock.so authfail parameters are not present in the pam files
+  ansible.builtin.lineinfile:
+    path: '{{ item }}'
+    backrefs: true
+    regexp: (.*pam_faillock.so authfail).*$
+    line: \1
+    state: present
   loop:
-    - system-auth
-    - password-auth
+    - /etc/pam.d/system-auth
+    - /etc/pam.d/password-auth
+  when:
+    - result_faillock_conf_check.stat.exists
+    - result_pam_faillock_enabled.found
+
+- name: Ensure pam_faillock.so is properly enabled using authconfig tool
+  ansible.builtin.command:
+    cmd: authconfig --enablefaillock --update
+  when:
+    - result_authconfig_check.stat.exists
```

Comment:
When testing on RHEL7 with Ansible, I get this:
```
INFO - xccdf_org.ssgproject.content_rule_accounts_passwords_pam_faillock_deny
INFO - Script pam_faillock_conflicting_settings.fail.sh using profile (all) OK
ERROR - Ansible playbook remediation run has exited with return code 2 instead of expected 0
```

```
TASK [Ensure pam_faillock.so is properly enabled using authselect] ********************************************************************************************
fatal: [192.168.122.12]: FAILED! => {"changed": false, "cmd": "authselect enable-feature with-faillock", "msg": "[Errno 2] No such file or directory", "rc": 2}
```

Reply:
There are a few other errors:
```
INFO - Script pam_faillock_disabled.fail.sh using profile (all) OK
ERROR - Ansible playbook remediation run has exited with return code 2 instead of expected 0
ERROR - The remediation failed for rule 'xccdf_org.ssgproject.content_rule_accounts_passwords_pam_faillock_deny'.
INFO - Script pam_faillock_expected_faillock_conf.pass.sh using profile (all) OK
INFO - Script pam_faillock_expected_pam_files.pass.sh using profile (all) OK
INFO - Script pam_faillock_lenient_faillock_conf.fail.sh using profile (all) OK
ERROR - Ansible playbook remediation run has exited with return code 2 instead of expected 0
ERROR - The remediation failed for rule 'xccdf_org.ssgproject.content_rule_accounts_passwords_pam_faillock_deny'.
INFO - Script pam_faillock_lenient_pam_files.fail.sh using profile (all) OK
INFO - Script pam_faillock_multiple_pam_unix_faillock_conf.fail.sh using profile (all) OK
INFO - Script pam_faillock_multiple_pam_unix_pam_files.fail.sh using profile (all) OK
ERROR - Script pam_faillock_not_required_pam_files.fail.sh using profile (all) found issue:
ERROR - Rule evaluation resulted in pass, instead of expected fail during initial stage 
ERROR - The initial scan failed for rule 'xccdf_org.ssgproject.content_rule_accounts_passwords_pam_faillock_deny'.
INFO - Script pam_faillock_stricter_faillock_conf.pass.sh using profile (all) OK
INFO - Script pam_faillock_stricter_pam_files.pass.sh using profile (all) OK
```

Reply:
Tested locally with Bash and Ansible remediation in RHEL7 and RHEL8 VMs. Also tested Bash and Ansible remediation in a RHEL8 container. All fine now.